bundles/org.openhab.binding.nobohub/src/main/java/org/openhab/binding/nobohub/internal/discovery/NoboHubDiscoveryService.java

```diff
@@ -0,0 +1,162 @@
+        }
+    }
+
+    @Override
+    public @Nullable ThingHandler getThingHandler() {
+        return hubBridgeHandler;
+    }
+
+    private final Runnable scanner = new Runnable() {
+        @Override
+        public void run() {
+            boolean found = false;
+            logger.info("Detecting Glen Dimplex Nobø Hubs, trying Multicast");
+            try {
+                MulticastSocket socket = new MulticastSocket(NOBO_HUB_MULTICAST_PORT);
+                found = waitOnSocket(socket, "multicast");
+            } catch (IOException ioex) {
+                logger.error("Failed detecting Nobø Hub multicast", ioex);
+            }
+
+            if (!found) {
+                logger.info("Detecting Glen Dimplex Nobø Hubs, trying Broadcast");
+
+                try {
+                    DatagramSocket socket = new DatagramSocket(NOBO_HUB_BROADCAST_PORT,
+                            InetAddress.getByName(NOBO_HUB_BROADCAST_ADDRESS));
+                    found = waitOnSocket(socket, "broadcast");
+                } catch (IOException ioex) {
+                    logger.error("Failed detecting Nobø Hub broadcast", ioex);
+                }
+            }
+        }
+
+        private boolean waitOnSocket(DatagramSocket socket, String type) throws IOException {
+            try {
+                socket.setBroadcast(true);
+
+                byte[] buffer = new byte[1024];
+                DatagramPacket data = new DatagramPacket(buffer, buffer.length);
+                String received = "";
+                while (!received.startsWith("__NOBOHUB__")) {
+                    socket.setSoTimeout((int) Duration.ofSeconds(4).toMillis());
+                    socket.receive(data);
+                    received = new String(buffer, 0, data.getLength());
+                }
+
+                logger.debug("Hub detection {}}: Received: {} from {}", type, received, data.getAddress());
+
+                String parts[] = received.split("__", 3);
+                if (3 != parts.length) {
+                    logger.debug("Data error, didn't contain three parts: '{}''", String.join("','", parts));
+                    return false;
+                }
+
+                String serialNumberStart = parts[parts.length - 1];
+                addDevice(serialNumberStart, data.getAddress().getHostName());
+                return true;
+            } finally {
+                socket.close();
+            }
+        }
+
+        private void addDevice(String serialNumberStart, String hostName) {
+            ThingUID bridge = new ThingUID(THING_TYPE_HUB, serialNumberStart);
+            String label = "Nobø Hub " + serialNumberStart;
+
+            Map<String, Object> properties = new HashMap<>(1);
+            properties.put("serialNumber", serialNumberStart);
+            properties.put("name", label);
+            properties.put("vendor", VENDOR);
+            properties.put("hostName", hostName);
```

Comment:
Please consider constants to ensure consistency.

Reply:
Fixed